Adds packaging docs on readthedocs (#1062)
citusdata/packaging: source/package-build-environment.rst

======================================

Introduction to Packaging Environments
--------------------------------------

In our packaging process, we utilize Docker images for each OS/release/PostgreSQL version combination. Docker images offer a reliable and consistent environment for packaging software applications. They provide several advantages for the packaging process:

1. **Isolation**: Docker containers encapsulate applications and their dependencies, ensuring isolation from the underlying host system. This isolation helps prevent conflicts between different software components and provides a clean and reproducible environment for packaging.

2. **Reproducibility**: Docker environments enable the creation of reproducible packaging environments. By defining the dependencies and configurations within a Docker image, it becomes easier to recreate the same environment across different systems, ensuring consistent packaging results.

3. **Portability**: Docker environments are portable and can be easily shared across different systems. This portability ensures that the same packaging results can be achieved across different systems, eliminating environment-related issues.

> 4. **Scalability**: Docker environments can be easily scaled up or down to meet the packaging requirements. This scalability ensures that the packaging process can be efficiently adjusted to handle different workload sizes.
> 
> 5. **Other Benefits**: Docker environments provide a secure, reliable, consistent, flexible, efficient, simple, and maintainable packaging environment. Docker isolates the packaging process from the underlying host system, ensuring a controlled and stable environment.
> 

Reviewer: I feel like those two are not something that we really benefit from, we can remove them altogether?

```suggestion
```

Docker Images for Various Package Types
---------------------------------------

> We utilize Docker images for each OS/release/PostgreSQL version combination to create a reproducible and consistent packaging environment.
> 
> For Debian-based packages, we can use the same Docker image for all PostgreSQL versions.
> 
> For example, the Debian/10 Docker image can be used for both PostgreSQL 13 and PostgreSQL 14 packages.
> 
> However, for RPM-based packages, we need a dedicated Docker image for each PostgreSQL version.
> 
> For instance, we require a Docker image for CentOS/7 with PostgreSQL 13 and another Docker image for CentOS/7 with PostgreSQL 14 packages.

Reviewer: ```suggestion
We utilize Docker images for each OS/release/PostgreSQL version combination to create a reproducible and consistent packaging environment.

For Debian-based packages, we use the same Docker image for all PostgreSQL versions.

For example, the Debian/10 Docker image is used for both PostgreSQL 13 and PostgreSQL 14 packages.

However, for RPM-based packages, we need a dedicated Docker image for each PostgreSQL version.

For instance, we require a Docker image for CentOS/7 with PostgreSQL 13 and another Docker image for CentOS/7 with PostgreSQL 14 packages.
```
